train.py

```diff
@@ -37,51 +38,74 @@
 X_test = zero_pad(X_test, SEQUENCE_LENGTH)
 
 # Different placeholders
-batch_ph = tf.placeholder(tf.int32, [None, SEQUENCE_LENGTH])
-target_ph = tf.placeholder(tf.float32, [None])
-seq_len_ph = tf.placeholder(tf.int32, [None])
-keep_prob_ph = tf.placeholder(tf.float32)
-
-# Embedding layer
-embeddings_var = tf.Variable(tf.random_uniform([vocabulary_size, EMBEDDING_DIM], -1.0, 1.0), trainable=True)
-batch_embedded = tf.nn.embedding_lookup(embeddings_var, batch_ph)
-
-# (Bi-)RNN layer(-s)
-rnn_outputs, _ = bi_rnn(GRUCell(HIDDEN_SIZE), GRUCell(HIDDEN_SIZE),
-                        inputs=batch_embedded, sequence_length=seq_len_ph, dtype=tf.float32)
-# rnn_outputs, _ = rnn(GRUCell(hidden_size), inputs=batch_embedded, sequence_length=seq_len_ph, dtype=tf.float32)
+batch_ph = tf.placeholder(tf.int32, [None, SEQUENCE_LENGTH],name='batch_ph')
+target_ph = tf.placeholder(tf.float32, [None],name='target_ph')
+seq_len_ph = tf.placeholder(tf.int32, [None],name='seq_len_ph')
+keep_prob_ph = tf.placeholder(tf.float32,name='keep_prob_ph')
+
+with tf.name_scope('Encoder'):
```

Comment:
I don't actually understand where you find encoder and decoder in this architecture. It doesn't work like encoder-decoder, because we don't generate any sequence, we just solve a classification task. Tell me if I'm wrong.

Reply:
I think the 'decoder' scope will definitely confuse a lot of people, so it shouldn't be here. Wrap two scopes (embedding and rnn) into 'encoder' doesn't seem necessary for me, but this scope may stay if you find it helpful